### DISCUSSION TITLE
upgrading to Masa using branches. Is this possible?

### DISCUSSION THREAD

**Discussion by wbssystems** [MasaCMS/MasaCMS#164] [General]
Is it possible create a new branch in our codebase, say called Masa7.4, and then run the core update so that the new masa core files get created just in that branch? Is it then possible to keep merging changes to master into this branch? So affectively i'd like to be able to switch branches which would give me wither Masa and Mura? Once the Masa branch was tested we would then merge into master and deploy.

**Comment by guustnieuwenhuis** [comment]
> Is it possible create a new branch in our codebase, say called Masa7.4, and then run the core update so that the new masa core files get created just in that branch?

Yes

> Is it then possible to keep merging changes to master into this branch?

Yes, assuming these changes are to themes, modules, plugins, etc. not Masa CMS code

> So affectively i'd like to be able to switch branches which would give me wither Masa and Mura?

I would suggest setting up 2 environments locally, once for each branch and NOT share the database. There are (small) database changes in Masa CMS. By separating the two environments, you'll have the best test case.

**Reply by wbssystems** [reply]
Hi 

Thanks for your reply.

I've successfully upgraded to Masa on a new 'masa7.4' branch off our master branch. I'm currently testing this in a local docker environment with a local copy of the production database. I can now successfully switch between branches(and therefore Masa and Mura codebases, with a Docker restart in-between). My plan is to hopefully fix issues with our site in Masa (i've noticed a few inline editing problems) on the Masa branch, and then periodically merge in any templates changes on the master branch into it.

What are the 'small' database changes made by Masa? Could there be issues if you have an instance of Mura and Masa both using the same remote DB for instance?
